kinvolk/inspektor-gadget
# Implement filetop gadget

This PR adds a new new filetop gadget from BCC. In this case only the BPF program is reused the the control plane is reimplemented in golang. 

How to test:

Use the CLI with the different options provided. Generate IO operations in the interested pods to see some events. 

```
$ ./kubectl-gadget filetop 
```

TODO:
- [x] Better way to clear the screen
- [x] How to pass errors back to caller in the tracer package? 
- [x] Use pkg/types.Event 
- [x] Add json tags to event 
- [x] Pass parameters to the tracer (depends on https://github.com/kinvolk/inspektor-gadget/issues/420)

TODO (future PRs):
- [ ] Create guide 
- [ ] Pass mount ns map as map instead of string (once https://github.com/cilium/ebpf/issues/517 and https://github.com/cilium/ebpf/issues/515 are fixed)
- [ ] Investigate why the big endianess BPF object makes BTFGen to fail
